control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1081,7 +1081,7 @@ func IsStorageAndCSIManaged(hostedControlPlane *hyperv1.HostedControlPlane) bool
 	return true
 }
 
-func (r *HostedControlPlaneReconciler) reconcile(ctx context.Context, hostedControlPlane *hyperv1.HostedControlPlane, createOrUpdate upsert.CreateOrUpdateFN, releaseImageProvider, userReleaseImageProvider *imageprovider.SimpleReleaseImageProvider, infraStatus infra.InfrastructureStatus) error {
```

Comment:
can the definition of SimpleReleaseImageProvider/others be removed now?

Reply:
If I'm not wrong, `SimpleReleaseImageProvider` is the actual implementation of `ReleaseImageProvider` interface. I replaced with the interface just for mocking. 
So I would say the answer is "no"